File: php/phpfind/tests/FindFileTest.php
<?php declare(strict_types=1);
use PHPUnit\Framework\TestCase;

require_once __DIR__ . '/../src/autoload.php';

use phpfind\FileType;
use phpfind\FindFile;

class FindFileTest extends TestCase
{
    public function test_findfile_abs_path()
    {
        $path = '/Users/cary/src/xfind/php/phpfind/src';
        $filename = 'findfile.php';
        $findfile = new FindFile($path, $filename, FileType::Code);
        $this->assertEquals('/Users/cary/src/xfind/php/phpfind/src/findfile.php',
            $findfile->filepath());
    }

    public function test_findfile_rel_path1()
    {
        $path = '.';
        $filename = 'findfile.php';
        $findfile = new FindFile($path, $filename, FileType::Code);
        $this->assertEquals('./findfile.php', $findfile->filepath());
    }

    public function test_findfile_rel_path2()
    {
        $path = './';
        $filename = 'findfile.php';
        $findfile = new FindFile($path, $filename, FileType::Code);
        $this->assertEquals('./findfile.php', $findfile->filepath());
    }

    public function test_findfile_rel_path3()
    {
        $path = '..';
        $filename = 'findfile.php';
        $findfile = new FindFile($path, $filename, FileType::Code);
        $this->assertEquals('../findfile.php', $findfile->filepath());
    }
}

File: cpp/cppfind/tests/FindFileTest.cpp
#include <catch2/catch.hpp>
#include "FindFile.h"

TEST_CASE("Verify fullpath findfile string equals expected", "[FindFile]") {
    std::string path = "/Users/cary/src/xfind/cpp/cppfind/src";
    std::string filename = "Finder.cpp";
    auto *findfile = new cppfind::FindFile(path, filename, cppfind::FileType::CODE);

    REQUIRE(findfile->path() == "/Users/cary/src/xfind/cpp/cppfind/src");
    REQUIRE(findfile->filename() == "Finder.cpp");
    REQUIRE(findfile->filetype() == cppfind::FileType::CODE);
    REQUIRE(findfile->string() == "/Users/cary/src/xfind/cpp/cppfind/src/Finder.cpp");
}

TEST_CASE("Verify relative path findfile string equals expected", "[FindFile]") {
    std::string path = ".";
    std::string filename = "Finder.cpp";
    auto *findfile = new cppfind::FindFile(path, filename, cppfind::FileType::CODE);

    REQUIRE(findfile->path() == ".");
    REQUIRE(findfile->filename() == "Finder.cpp");
    REQUIRE(findfile->filetype() == cppfind::FileType::CODE);
    REQUIRE(findfile->string() == "./Finder.cpp");
}
